test/inductor/test_custom_lowering.py

```diff
@@ -0,0 +1,140 @@
+# Owner(s): ["module: inductor"]
+
+import torch
+
+from torch._inductor.ir import Pointwise
+from torch._inductor.lowering import register_lowering
+from torch._inductor.virtualized import ops
+
+from torch.testing._internal.common_utils import TestCase as TorchTestCase
+from torch.testing._internal.inductor_utils import HAS_CPU, HAS_CUDA
+
+
+# These tests check issues for lowerings that aren't in the main pytorch repo
+class TestCustomLowering(TorchTestCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.test_inductor_ops = torch.library.Library("test_inductor_ops", "DEF")
+        cls.impl_cuda = torch.library.Library("test_inductor_ops", "IMPL", "CUDA")
+        cls.impl_meta = torch.library.Library("test_inductor_ops", "IMPL", "Meta")
+        cls._register_jagged_to_padded_dense()
+
+    @classmethod
+    def tearDown(cls):
+        super().tearDownClass()
+
+    @classmethod
+    def _register_jagged_to_padded_dense(cls):
+        # Approximation of fbgemm.jagged_to_padded_dense_forward
+        cls.test_inductor_ops.define(
+            "jagged_to_padded_dense(Tensor input, Tensor offsets, SymInt max_seq_len) -> Tensor"
+        )
+
+        def j2pd_meta(inp, offsets, max_seq_len):
+            return torch.empty(
+                (offsets.shape[0] - 1, max_seq_len, inp.shape[1]),
+                device=inp.device,
+                dtype=inp.dtype,
+            )
+
+        def j2pd_cuda(inp, offsets, max_seq_len):
+            res = torch.zeros(
+                (offsets.shape[0] - 1, max_seq_len, inp.shape[1]),
+                device=inp.device,
+                dtype=inp.dtype,
+            )
+            for b in range(offsets.shape[0] - 1):
+                for r in range(offsets[b + 1] - offsets[b]):
+                    res[b][r] = inp[offsets[b] + r]
+            return res
+
+        def j2pd_lowering(inp, offsets, max_seq_len):
+            offsets_loader = offsets.make_loader()
+            inp_loader = inp.make_loader()
+            jagged_len = inp.get_size()[0]
+            offsets_dtype = offsets.get_dtype()
+
+            def inner_fn(index):
+                batch_idx, seq_idx, emb_idx = index
+
+                begin_idx = ops.indirect_indexing(
+                    offsets_loader([batch_idx]),
+                    jagged_len + 1,
+                )
+                end_idx = offsets_loader([batch_idx + 1])
+                jagged_idx = begin_idx + seq_idx
+
+                return ops.masked(
+                    ops.lt(
+                        ops.index_expr(jagged_idx, offsets_dtype),
+                        end_idx,
+                    ),
+                    lambda: inp_loader([jagged_idx, emb_idx]),
```

Comment:
Feels to me like the loader for a zero-size tensor should just return a constant NaN or something. Emitting a load instruction that will always be masked out doesn't make sense to me.

Reply:
@peterbell10 after thinking about your comment for a bit, I still feel like this is a reasonable fix - I put more details/discussion in #107066 if you have any other comments.

I agree that yes, ideally we should convert this load to a constant/tl.full/etc. However, I think it would be a much more involved change; and I suspect the perf difference will be pretty minimal (but I'll double check perf once I get tests passing)

Reply:
The change seems relatively simple to me, see #107074.